test/ICSharpCode.SharpZipLib.Tests/Base/InflaterDeflaterTests.cs

```diff
@@ -89,35 +118,73 @@ private MemoryStream Deflate(byte[] data, int level, bool zlib)
 			return memoryStream;
 		}
 
-		private void RandomDeflateInflate(int size, int level, bool zlib)
+		private async Task<MemoryStream> DeflateAsync(byte[] data, int level, bool zlib)
+		{
+			var memoryStream = new MemoryStream();
+
+			var deflater = new Deflater(level, !zlib);
+			using (DeflaterOutputStream outStream = new DeflaterOutputStream(memoryStream, deflater))
+			{
+				outStream.IsStreamOwner = false;
+				await outStream.WriteAsync(data, 0, data.Length);
+				await outStream.FlushAsync();
+				outStream.Finish();
+			}
+			return memoryStream;
+		}
+
+		private static byte[] GetRandomTestData(int size)
 		{
 			byte[] buffer = new byte[size];
 			var rnd = new Random();
 			rnd.NextBytes(buffer);
 
+			return buffer;
+		}
+
+		private void RandomDeflateInflate(int size, int level, bool zlib)
+		{
+			byte[] buffer = GetRandomTestData(size);
+
 			MemoryStream ms = Deflate(buffer, level, zlib);
 			Inflate(ms, buffer, level, zlib);
 		}
 
+		private async Task RandomDeflateInflateAsync(int size, int level, bool zlib)
+		{
+			byte[] buffer = GetRandomTestData(size);
+
+			MemoryStream ms = await DeflateAsync(buffer, level, zlib);
+			await InflateAsync(ms, buffer, level, zlib);
+		}
+
 		/// <summary>
 		/// Basic inflate/deflate test
 		/// </summary>
 		[Test]
 		[Category("Base")]
-		public void InflateDeflateZlib()
+		public void InflateDeflateZlib([Range(0, 9)] int level)
 		{
-			for (int level = 0; level < 10; ++level)
-			{
-				RandomDeflateInflate(100000, level, true);
-			}
+			RandomDeflateInflate(100000, level, true);
+		}
+
+		/// <summary>
+		/// Basic async inflate/deflate test
+		/// </summary>
+		[Test]
+		[Category("Base")]
+		[Category("Async")]
+		public async Task InflateDeflateZlibAsync([Range(0, 9)] int level)
+		{
+			await RandomDeflateInflateAsync(100000, level, true);
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/piksel/SharpZipLib/pullRequest?prid=5745476)